apps/meteor/server/methods/deleteUser.ts

```diff
@@ -15,45 +15,56 @@ declare module '@rocket.chat/ddp-client' {
 	}
 }
 
-Meteor.methods<ServerMethods>({
-	async deleteUser(userId, confirmRelinquish = false) {
-		check(userId, String);
-		const uid = Meteor.userId();
-		if (!uid || (await hasPermissionAsync(uid, 'delete-user')) !== true) {
-			throw new Meteor.Error('error-not-allowed', 'Not allowed', {
-				method: 'deleteUser',
-			});
-		}
+export const executeDeleteUser = async (fromUserId: IUser['_id'], userId: IUser['_id'], confirmRelinquish = false): Promise<boolean> => {
+	if ((await hasPermissionAsync(fromUserId, 'delete-user')) !== true) {
```

Comment:
move the permission check to the method scope.. the API already checks permission